integration/findandmodify_compat_test.go

```diff
@@ -422,34 +422,47 @@ func TestFindAndModifyCompatUpdateCurrentDate(t *testing.T) {
 				{"query", bson.D{{"_id", "datetime"}}},
 				{"update", bson.D{{"$currentDate", 1}}},
 			},
+			providers:  []shareddata.Provider{shareddata.DateTimes},
```

Comment:
Could you enlighten why we only test for datetime provider? We should be testing for all providers.

Reply:
We are having a filter of `{_id: "datetime"}` in these test cases. What is use of running it against all the providers? We will be reducing test execution time by limiting the providers whenever we use a specific `_id` filter in test cases.

Reply:
The reason why we run all providers is often one of providers catch some issue, particularly related to array filters. You are right, for this case datetime provider should be enough